samza-yarn/src/main/java/org/apache/samza/job/yarn/LocalizerResourceMapper.java

```diff
@@ -0,0 +1,114 @@
+  private Map<String, LocalResource> localResourceMap;
+
+  public LocalizerResourceMapper(Config config, YarnConfiguration yarnConfiguration) {
+    this.config = config;
+    this.yarnConfiguration = yarnConfiguration;
+    this.localResourceMap = new HashMap<>();
+  }
+
+  public void map() throws LocalizerResourceException {
+    // filter to get localizer resource config
+    Config localizerResourceConfig;
+    try {
+      YarnConfig yarnConfig = new YarnConfig(config);
+      localizerResourceConfig = yarnConfig.getLocalizerResourceConfigs();
+      log.info("localizerResourceConfig: {} ", localizerResourceConfig);
+    } catch (SamzaException e) {
+      log.info("localizerResourceConfig not found in config");
+      return;
+    }
+
+    // create LocalResource and put in map
+    for (Map.Entry<String, String> entry : localizerResourceConfig.entrySet()) {
+      LocalizerResourceConfigParser resourceConfigParser = new LocalizerResourceConfigParser(entry.getKey(), entry.getValue());
+
+      if (!resourceConfigParser.isValid()) {
+        throw new LocalizerResourceException("resourceConfigParser fails. Resource visibility, type, name or path is missing or invalid for " + entry);
+      }
+
+      Path resourcePath = resourceConfigParser.getResourcePath();
+
+      LocalResource localResource = createLocalResource(resourcePath, resourceConfigParser.getResourceType(), resourceConfigParser.getResourceVisibility());
+
+      localResourceMap.put(resourceConfigParser.getResourceName(), localResource);
+      log.info("preparing local resource: {}", resourceConfigParser.getResourceName());
+    }
+
+  }
+
+  private LocalResource createLocalResource(Path resourcePath, LocalResourceType resourceType, LocalResourceVisibility resourceVisibility) throws LocalizerResourceException {
+    LocalResource localResource = Records.newRecord(LocalResource.class);
+    URL resourceUrl = ConverterUtils.getYarnUrlFromPath(resourcePath);
+    try {
+      FileStatus resourceFileStatus = resourcePath.getFileSystem(yarnConfiguration).getFileStatus(resourcePath);
+
+      if (null == resourceFileStatus) {
+        throw new LocalizerResourceException("Check getFileStatus implementation. getFileStatus gets unexpected null for resourcePath " + resourcePath);
+      }
+
+      localResource.setResource(resourceUrl);
+      log.info("setLocalizerResource for {}", resourceUrl);
+      localResource.setSize(resourceFileStatus.getLen());
+      localResource.setTimestamp(resourceFileStatus.getModificationTime());
+      localResource.setType(resourceType);
+      localResource.setVisibility(resourceVisibility);
+      return localResource;
+    } catch (IOException ioe) {
+      log.error("IO Exception when accessing the resource file status from the filesystem", ioe);
+      throw new LocalizerResourceException("IO Exception when accessing the resource file status from the filesystem");
+    }
+
+  }
+
+  public Map<String, LocalResource> getResourceMap() {
```

Comment:
I have put the map() function into the constructor, and now the caller just need to initialize the  object and then call getResourceMap(). I think it is simple enough. Although we can make it one static function, but that is not very testing friendly. 
Regarding the immutable map, Thanks for the suggestion, I changed it to ImmutableMap.